### PR TITLE
Update log for tls socket

### PR DESCRIPTION
**Pull request by hihigupt**
https://github.com/Azure-Samples/iot-middleware-freertos-samples/issues/44

### CODE REVIEW

File: demos/common/transport/transport_tls_socket_using_mbedtls.c [hunk 1]
```diff
@@ -830,7 +830,7 @@ int32_t TLS_Socket_Recv( NetworkContext_t * pxNetworkContext,
         ( lTLSStatus == MBEDTLS_ERR_SSL_WANT_WRITE ) )
     {
         LogDebug( ( "Failed to read data. However, a read can be retried on this error. "
-                    "mbedTLSError= %s : %s.",
+                    "mbedTLSError[%d]= %s : %s.", lTLSStatus,
```
Comment (danewalton) [inline demos/common/transport/transport_tls_socket_using_mbedtls.c:833]: nit: why have this be basetype_t in the disconnect but int32_t here?

**Review by danewalton — APPROVED** [review]

File: demos/common/transport/transport_tls_socket_using_mbedtls.c (hunk 1 above)
Comment (hihigupt) [reply demos/common/transport/transport_tls_socket_using_mbedtls.c:833]: Issue in source where it is taken from. For consistency I will make it int32_t.

File: demos/common/transport/transport_tls_socket_using_mbedtls.c [hunk 2]
```diff
@@ -840,8 +840,8 @@ int32_t TLS_Socket_Recv( NetworkContext_t * pxNetworkContext,
     }
     else if( lTLSStatus < 0 )
     {
-        LogError( ( "Failed to read data: mbedTLSError= %s : %s.",
-                    mbedtlsHighLevelCodeOrDefault( lTLSStatus ),
+        LogError( ( "Failed to read data: mbedTLSError[%d]= %s : %s.",
+                    lTLSStatus, mbedtlsHighLevelCodeOrDefault( lTLSStatus ),
```
Comment (jspaith) [inline demos/common/transport/transport_tls_socket_using_mbedtls.c:844]: Nit - why is this function this called `lTLSStatus` but other places its `lMbedtlsError`?  It seems it should be consistent??

**Review by jspaith — APPROVED** [review]

File: demos/common/transport/transport_tls_socket_using_mbedtls.c (hunk 2 above)
Comment (hihigupt) [reply demos/common/transport/transport_tls_socket_using_mbedtls.c]: Ack

**Pull request merged**